src/components/MAPI/apps/ClusterDetailAppListWidgetUninstall.tsx

```diff
@@ -0,0 +1,144 @@
+import { useAuthProvider } from 'Auth/MAPI/MapiAuthProvider';
+import { Box, Text } from 'grommet';
+import ErrorReporter from 'lib/errors/ErrorReporter';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import { useHttpClient } from 'lib/hooks/useHttpClient';
+import { extractErrorMessage } from 'MAPI/utils';
+import * as applicationv1alpha1 from 'model/services/mapi/applicationv1alpha1';
+import React, { useState } from 'react';
+import { mutate } from 'swr';
+import Button from 'UI/Controls/Button';
+import ConfirmationPrompt from 'UI/Controls/ConfirmationPrompt';
+import ClusterIDLabel, {
+  ClusterIDLabelType,
+} from 'UI/Display/Cluster/ClusterIDLabel';
+import LoadingPlaceholder from 'UI/Display/LoadingPlaceholder/LoadingPlaceholder';
+import ClusterDetailAppListWidget from 'UI/Display/MAPI/apps/ClusterDetailAppListWidget';
+
+import { deleteAppWithName } from './utils';
+
+interface IClusterDetailAppListWidgetUninstallProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof ClusterDetailAppListWidget>,
+    'title'
+  > {
+  app?: applicationv1alpha1.IApp;
+  onAppUninstalled?: () => void;
+}
+
+const ClusterDetailAppListWidgetUninstall: React.FC<IClusterDetailAppListWidgetUninstallProps> = ({
+  app,
+  onAppUninstalled,
+  ...props
+}) => {
+  const auth = useAuthProvider();
+  const appClient = useHttpClient();
+
+  const uninstallApp = async () => {
+    if (!app) return;
+
+    try {
+      await deleteAppWithName(
+        appClient,
+        auth,
+        app.metadata.namespace!,
+        app.metadata.name
+      );
+
+      mutate(
+        applicationv1alpha1.getAppListKey({ namespace: app.metadata.namespace })
+      );
```

Comment:
One problem with this: after deleting an app, the deleted app won't be updated until fetching the app list again. Which means that on slower internet connections, the user will still see the unchanged app in the list after deleting it.

We can update the cache manually, to get instant feedback in the UI.

Here's an example of how we do this in other places:
https://github.com/giantswarm/happa/blob/5912045903bbf35f01e33633cf6c79b10136fccb/src/components/MAPI/clusters/ClusterDetail/utils.ts#L51

In this example we mutate the cluster resource manually, adding its `deletionTimestamp` property value (which makes it be displayed as deleted), and then we update the local cache with the updated value.